src/services/address.service.ts

```diff
@@ -38,35 +38,98 @@ class AddressService {
+            this.request(addressRequest)
+            .then((response) => {
+                if(response.length < 2) {
+                    loggerService.warning({ path: "/address/distance", message: "Search conditions did not match to at least 2 addresses." }).flush();
+                    reject("Search conditions must match to at least 2 addresses");
+                }
+
+                const address1 = response[0];
+                const address2 = response[1];
+
+                var lat1 = address1["latitude"];
+                var long1 = address1["longitude"];
+                var lat2 = address2["latitude"];
+                var long2 = address2["longitude"];
+
+                var dist = this.getDistance(lat1, long1, lat2, long2);
+                var distKM = dist;
+                var distMI = dist * 0.621;
+
+                resolve({
+                    "kilometers": distKM,
+                    "miles": distMI
+                });
+            })
+            .catch((err) => {
+                loggerService.error({ path: "/address/distance", message: `${(err as Error).message}` }).flush();
+                reject(err);
+            })
+        });
     }
 
-    // private async getDistance(lat1: string, lon1: string, lat2: string, lon2: string) {
-    //     // Defining this function inside of this private method means it's
-    //     // not accessible outside of it, which is perfect for encapsulation.
-    //     const toRadians = (degrees: string) => {
-    //         return degrees * (Math.PI / 180);
-    //     }
+    public async getcity(addressRequest?: any): Promise<any> {
+        return new Promise<any>(async (resolve, reject) => {
+            if(addressRequest.body["zipcode"] == null || addressRequest.body.length > 1) {
+                loggerService.warning({ path: "/address/getcity", message: "Bad input. Only zipcode parameter should be used." }).flush();
+                reject("Getcity requires a zipcode attribute. Do not include any other attributes.");
+            }
+            
+            this.request(addressRequest)
+            .then((response) => {
+                if(response.length == 0) {
+                    loggerService.warning({ path: "/address/getcity", message: "No city was found for the zipcode provided."}).flush();
+                    reject("No city was found for this zipcode.");
+                }
 
-    //     // Radius of the Earth in KM
-    //     const R = 6371;
+                const city = response[0]["city"];
+                const state = response[0]["state"];
 
-    //     // Convert Lat and Longs to Radians
-    //     const dLat = toRadians(lat2 - lat1);
-    //     const dLon = toRadians(lon2 - lon1);
+                resolve({
+                    "city": city + ", " + state,
+                });
+            })
+            .catch((err) => {
+                loggerService.error({ path: "/address/getcity", message: `${(err as Error).message}` }).flush();
```

Comment:
I wonder - is this really an error for this app or would it just be a warning? Errors are usually used when the current app is crashing or about to crash.